### PR TITLE
chore(gatsby, gatsby-transformer-remark, gatsby-plugin-sharp): Cleanup v4 compile flag conditions

### PR DESCRIPTION
**Pull request by marvinjude**
## Description

In preparation for v5 canaries, This PR makes codepaths behind `_CFLAGS_.GATSBY_MAJOR === 4` conditions the default and also removes non `_CFLAGS_.GATSBY_MAJOR === 4` codepaths.

[sc-54975]

### CODE REVIEW

File: packages/gatsby-plugin-sharp/src/gatsby-node.js [hunk 1]
```diff
@@ -131,16 +126,9 @@ exports.onPostBootstrap = async ({ reporter, cache, store }) => {
...
-  } else {
-    exports.unstable_onPluginInit = async ({ actions }, pluginOptions) => {
-      setActions(actions)
-      setPluginOptions(pluginOptions)
-    }
+  exports.onPluginInit = async ({ actions }, pluginOptions) => {
+    setActions(actions)
+    setPluginOptions(pluginOptions)
   }
 }
```
Comment (wardpeet) [inline packages/gatsby-plugin-sharp/src/gatsby-node.js:133]: This file can only contain
```
exports.onPluginInit = async ({ actions }, pluginOptions) => {
    setActions(actions)
    setPluginOptions(pluginOptions)
  }
```

File: packages/gatsby/src/commands/serve.ts [hunk 2]
```diff
@@ -186,132 +186,128 @@ module.exports = async (program: IServeProgram): Promise<void> => {
...
-      // TODO: Handle case of engine not being generated
-    }
+      }
+      return next()
+    })
+  } catch (error) {
+    // TODO: Handle case of engine not being generated
   }
 
+  // end
```
Comment (wardpeet) [inline packages/gatsby/src/commands/serve.ts:310]:
```suggestion
```

**Review by wardpeet — COMMENTED** [review]
Looks good, small remarks.

File: packages/gatsby-plugin-sharp/src/gatsby-node.js (hunk 1 above)
Comment (marvinjude) [reply packages/gatsby-plugin-sharp/src/gatsby-node.js:133]: Do you removing `if (coreSupportsOnPluginInit) {` check ?

Comment (wardpeet) [reply packages/gatsby-plugin-sharp/src/gatsby-node.js:133]: Yes

Comment (marvinjude) [reply packages/gatsby-plugin-sharp/src/gatsby-node.js:133]: 👍

**Review by wardpeet — APPROVED** [review]

**Pull request merged**